packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift

```diff
@@ -230,30 +230,31 @@ public struct FlutterSerializedModel: Model, ModelIdentifiable, JSONValueHolder
                         continue
                     }
 
-                    result[key] = [
-                        "modelName": nextModelName,
-                        "serializedData": try generateSerializedData(
+                    let dataMap = try generateSerializedData(
```

Comment:
Similar comment to Hui's. Instead of merging `modelName` into the model (which may conflict with already-present data), it may be better to return a wrapped value or use `__typename` instead which is part of the GraphQL spec.